core/node/eth_sender/src/aggregator.rs

```diff
@@ -199,11 +241,68 @@ impl Aggregator {
             ready_for_execute_batches,
             last_sealed_l1_batch,
         )
-        .await;
+        .await?;
+
+        let start_index = self.config.priority_tree_start_index;
+        let priority_merkle_tree = self.get_or_init_tree().await;
+
+        let mut priority_ops_proofs = vec![];
+        for batch in l1_batches.iter() {
+            let priority_tree_start_index = if let Some(start_index) = start_index {
+                start_index
+            } else {
+                // Start index not present, we push empty proof.
+                priority_ops_proofs.push(Default::default());
+                continue;
+            };
+
+            let first_priority_op_id_option = match storage
+                .blocks_dal()
+                .get_batch_first_priority_op_id(batch.header.number)
+                .await
+                .unwrap()
+            {
+                // Batch has no priority ops, no proofs to send
+                None => None,
+                // We haven't started to use the priority tree in the contracts yet
+                Some(id) if id < priority_tree_start_index => None,
+                Some(id) => Some(id),
+            };
+
+            let count = batch.header.l1_tx_count as usize;
+            if let Some(first_priority_op_id_in_batch) = first_priority_op_id_option {
+                let new_l1_tx_hashes = storage
+                    .transactions_dal()
+                    .get_l1_transactions_hashes(
+                        priority_tree_start_index + priority_merkle_tree.length(),
+                    )
+                    .await
+                    .unwrap();
+                for hash in new_l1_tx_hashes {
+                    priority_merkle_tree.push_hash(hash);
+                }
+
+                priority_merkle_tree.trim_start(
```

Comment:
Nit: Could make sense to add a comment explaining trimming here. AFAIU, this trims the tree to retain just hashes from the newly processed L1 batch.